sklearn/kernel_approximation.py

```diff
@@ -444,6 +446,12 @@ class SkewedChi2Sampler(
     1.0
     """
 
+    _parameter_constraints = {
+        "skewedness": [Interval(Real, None, None, closed="neither")],
+        "n_components": [Interval(Integral, 0, None, closed="left")],
```

Comment:
```suggestion
        "n_components": [Interval(Integral, 1, None, closed="left")],
```

Having zero components will return an empty array of shape `(n_samples, 0)`. I don't see a use case where we don't want to raise an error. Any subsequent estimators will raise an error with such input.

Reply:
Nice catch.